packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2100/onlp/builds/src/module/src/ledi.c

```diff
@@ -0,0 +1,284 @@
+};
+
+static char file_names[][10] =  /* must map with onlp_led_id */
+{
+    "reserved",
+    "status",
+    "fan",
+    "psu1",
+    "psu2",
+	"uid"
+};
+
+/*
+ * Get the information for the given LED OID.
+ */
+static onlp_led_info_t linfo[] =
+{
+    { }, /* Not used */
+    {
+        { ONLP_LED_ID_CREATE(LED_SYSTEM), "Chassis LED 1 (SYSTEM LED)", 0 },
+        ONLP_LED_STATUS_PRESENT,
+        ONLP_LED_CAPS_ON_OFF | ONLP_LED_CAPS_GREEN | ONLP_LED_CAPS_GREEN_BLINKING |
+        ONLP_LED_CAPS_RED | ONLP_LED_CAPS_RED_BLINKING | ONLP_LED_CAPS_AUTO,
+    },
+    {
+        { ONLP_LED_ID_CREATE(LED_FAN), "Chassis LED 2 (FAN LED)", 0 },
+        ONLP_LED_STATUS_PRESENT,
+        ONLP_LED_CAPS_ON_OFF | ONLP_LED_CAPS_GREEN | ONLP_LED_CAPS_GREEN_BLINKING |
+        ONLP_LED_CAPS_RED | ONLP_LED_CAPS_RED_BLINKING | ONLP_LED_CAPS_AUTO,
+    },
+    {
+        { ONLP_LED_ID_CREATE(LED_PSU1), "Chassis LED 3 (PSU1 LED)", 0 },
+        ONLP_LED_STATUS_PRESENT,
+        ONLP_LED_CAPS_ON_OFF | ONLP_LED_CAPS_GREEN | ONLP_LED_CAPS_GREEN_BLINKING |
+        ONLP_LED_CAPS_RED | ONLP_LED_CAPS_RED_BLINKING | ONLP_LED_CAPS_AUTO,
+    },
+	{
+		{ ONLP_LED_ID_CREATE(LED_PSU2), "Chassis LED 4 (PSU2 LED)", 0 },
+		ONLP_LED_STATUS_PRESENT,
+		ONLP_LED_CAPS_ON_OFF | ONLP_LED_CAPS_GREEN | ONLP_LED_CAPS_GREEN_BLINKING |
+		ONLP_LED_CAPS_RED | ONLP_LED_CAPS_RED_BLINKING | ONLP_LED_CAPS_AUTO,
+	},
+	{
+		{ ONLP_LED_ID_CREATE(LED_UID), "Chassis LED 5 (UID LED)", 0 },
+		ONLP_LED_STATUS_PRESENT,
+		ONLP_LED_CAPS_ON_OFF | ONLP_LED_CAPS_BLUE | ONLP_LED_CAPS_BLUE_BLINKING |
+		ONLP_LED_CAPS_AUTO,
+	}
+};
+
+static int driver_to_onlp_led_mode(enum onlp_led_id id, char* driver_led_mode)
+{
+    int i, nsize = sizeof(led_map)/sizeof(led_map[0]);
+
+    for (i = 0; i < nsize; i++)
+    {
+        if (id == led_map[i].id &&
+            !strncmp(led_map[i].driver_led_mode, driver_led_mode, driver_value_len))
+        {
+            return led_map[i].onlp_led_mode;
+        }
+    }
+
+    return 0;
+}
+
+static char* onlp_to_driver_led_mode(enum onlp_led_id id, onlp_led_mode_t onlp_led_mode)
+{
+    int i, nsize = sizeof(led_map)/sizeof(led_map[0]);
+
+    for (i = 0; i < nsize; i++)
+    {
+        if (id == led_map[i].id && onlp_led_mode == led_map[i].onlp_led_mode)
+        {
+            return led_map[i].driver_led_mode;
+        }
+    }
+
+    return LED_MODE_OFF;
+}
+
+/*
+ * This function will be called prior to any other onlp_ledi_* functions.
+ */
+int
+onlp_ledi_init(void)
+{
+    /*
+     * TODO setting UI LED to off when it will be supported
+     */
+
+    return ONLP_STATUS_OK;
+}
+
+int
+onlp_ledi_info_get(onlp_oid_t id, onlp_led_info_t* info)
+{
+    int  len, local_id = 0;
+    uint8_t data[driver_value_len] = {0};
+    char fullpath[50] = {0};
+
+    VALIDATE(id);
+
+    local_id = ONLP_OID_ID_GET(id);
+
+    /* get fullpath */
+    snprintf(fullpath, sizeof(fullpath), "%s%s", prefix_path, file_names[local_id]);
```

Comment:
This pattern occurs so frequently that the `onlp_file_read()` function already handles it for you:
https://github.com/opencomputeproject/OpenNetworkLinux/blob/f77e29cd513b8cede557af820750d83c57985d00/packages/base/any/onlp/src/onlplib/module/inc/onlplib/file.h#L48

You don't need the `fullpath` temporary variable. You can call `onlp_file_read` as follows:

`onlp_file_read(data, sizeof(data), &len, "%s%s", prefix_path, file_names[local_id]`